model_analyzer/config/generate/model_variant_name_manager.py

```diff
@@ -21,95 +22,79 @@ class ModelVariantNameManager:
+        mcd = self._copy_and_restore_mcd_name(model_name, model_config_dict)
 
-    def _model_variant_exists(self, base_model_name, param_combo):
-        """ 
-        Returns true if a model variant name already exists for this combination 
-        """
-        hashable_key = self._get_hashable_key(param_combo)
+        variant_found, model_variant_name = self._find_existing_variant(
+            model_name, mcd)
 
-        return base_model_name in self._variant_names \
-           and hashable_key in self._variant_names[base_model_name]
+        if variant_found:
+            return (True, model_variant_name)
 
-    def _get_model_variant_name(self, base_model_name, param_combo):
-        """
-        Returns the model variant name that already exists for this combination
-        """
-        hashable_key = self._get_hashable_key(param_combo)
-        return self._variant_names[base_model_name][hashable_key]
+        if self._is_default_config(param_combo):
+            return (False, model_name + '_config_default')
 
-    def _make_model_variant_name(self, base_model_name, param_combo):
-        """
-        Create and save a new model variant for this base model
-        """
-        if param_combo == DEFAULT_CONFIG_PARAMS:
-            variant_name = f'{base_model_name}_config_default'
-        else:
-            model_name_index = self._get_next_model_name_index(base_model_name)
-            variant_name = f'{base_model_name}_config_{model_name_index}'
-            self._add_model_variant_name(base_model_name, param_combo,
-                                         variant_name)
-        return variant_name
-
-    def _add_model_variant_name(self, base_model_name, param_combo,
-                                variant_name):
-        """ 
-        Save a new model variant name for this base model
-        """
-        if base_model_name not in self._variant_names:
-            self._variant_names[base_model_name] = {}
+        model_variant_name = self._create_new_model_variant(model_name, mcd)
 
-        hashable_key = self._get_hashable_key(param_combo)
-        self._variant_names[base_model_name][hashable_key] = variant_name
+        return (False, model_variant_name)
 
-    def _get_next_model_name_index(self, base_model_name):
-        """
-        Get the next index to be used as a unique number for variant naming
-        """
-        if base_model_name not in self._model_name_index:
-            self._model_name_index[base_model_name] = 0
-        else:
-            self._model_name_index[base_model_name] += 1
+    def _copy_and_restore_mcd_name(self, model_name: str,
+                                   model_config_dict: Dict) -> Dict:
+        mcd = deepcopy(model_config_dict)
+        mcd['name'] = model_name
 
-        return self._model_name_index[base_model_name]
+        return mcd
 
-    def _get_hashable_key(self, obj):
-        obj_copy = deepcopy(obj)
+    def _find_existing_variant(self, model_name: str,
```

Comment:
Input parameter `model_name` is not used.

Reply:
Removed